src/main/java/org/templegalaxia/patterns/testing/RingIterator.java

```diff
@@ -0,0 +1,29 @@
+package org.templegalaxia.patterns.testing;
+
+import heronarts.lx.LX;
+import heronarts.lx.LXCategory;
+import heronarts.lx.color.LXColor;
+import heronarts.lx.model.LXPoint;
+import heronarts.lx.parameter.CompoundParameter;
+import org.templegalaxia.patterns.TemplePattern;
+
+@LXCategory("Test")
+public class RingIterator extends TemplePattern {
+  private final CompoundParameter ringNum =
+      new CompoundParameter("Petal", 0, 0, model.rings.size() - 1);
+
+  public RingIterator(LX lx) {
+    super(lx);
+    addParameter(ringNum);
+  }
+
+  public void run(double deltaMs) {
+    for (LXPoint p : model.points) {
+      colors[p.index] = LXColor.BLACK;
+    }
+
+    for (LXPoint p : model.rings.get((int) ringNum.getValuef()).points) {
+      colors[p.index] = LXColor.WHITE;
```

Comment:
where do these LXPoint.index() values get set?

Reply:
Oh weird it is a static counter on the LXPoint class that increments when LXPoint objects are initialized:

```

    public LXPoint(float var1, float var2, float var3) {
        this.xn = 0.0F;
        this.yn = 0.0F;
        this.zn = 0.0F;
        this.rn = 0.0F;
        this.x = var1;
        this.y = var2;
        this.z = var3;
        this.index = counter++;
        this.update();
    }
```

I don't know if it's likely to be the same order as our wiring... unless you are intentionally loading and initializing the points in the same order that they're going to be wired. Is that the case?